woocommerce/woocommerce
# Added check out DB update and run if it exists

### Changes proposed in this Pull Request:

This PR adds a check for the database update notice, as seen in the issue linked below. It also adds these two methods:

* `runDatabaseUpdate()`: Check if the notice exists, and if it does, run the database update
* `checkDatabaseUpdateComplete()`: Check if the update complete button shows and click it. Otherwise, refresh the page until the database update is complete.

Closes https://github.com/woocommerce/woocommerce/issues/30396

### How to test the changes in this Pull Request:

1. Make sure CI passes
2. Set up a test site (I used one external to the Docker environment) with a previous version of WooCommerce that will trigger the database update notice. In my testing, I used `5.5.2` and upgraded to the `nightly` version.
3. Run the update woocommerce script, pointing to your test site in nonheadless mode with a command like the one below:

```
SMOKE_TEST_URL="localhost" SMOKE_TEST_ADMIN_USER="test" SMOKE_TEST_ADMIN_PASSWORD="password" UPDATE_WC=1 npx wc-e2e test:e2e-dev ./tests/e2e/specs/smoke-tests/update-woocommerce.js
```

4. Watch the test run and verify the database update is started, and the test waits for it to complete before exiting.
